modules/components/src/AdvancedFacetView/index.js

```diff
@@ -39,6 +61,66 @@ export default class AdvancedFacetView extends React.Component {
   };
   constructFilterId = ({ field, value }) =>
     value ? `${field}---${value}` : field;
+
+  handleSqonChange = ({ sqon }) => {
+    const { onSqonFieldChange = () => {} } = this.props;
+    this.setState(
+      {
+        isLoading: true,
+      },
+      () => {
+        onSqonFieldChange({ sqon });
+      },
+    );
+  };
+
+  getSnapshotBeforeUpdate(prevProps, prevState) {
+    const aggChanged = !isEqual(
+      this.props.aggregations,
+      prevProps.aggregations,
+    );
+    const sqonChanged = !isEqual(this.props.sqon, prevProps.sqon);
+    return {
+      shouldEndLoading: aggChanged || sqonChanged,
+    };
+  }
+
+  componentDidUpdate(prevProps, prevState, { shouldEndLoading }) {
+    const shouldRecomputeDisplayTree = !isEqual(
+      {
+        elasticMapping: this.props.elasticMapping,
+        extendedMapping: this.props.extendedMapping,
+      },
+      {
+        elasticMapping: prevProps.elasticMapping,
+        extendedMapping: prevProps.extendedMapping,
+      },
```

Comment:
not really important, but could use lodash `pick` for this type of object projection